backend/src/impl/default_controllers_impl.py

```diff
@@ -150,12 +160,52 @@ def benchmarkconfigs_get(parent: Optional[str]) -> list[BenchmarkConfig]:
 def benchmark_benchmark_id_get(benchmark_id: str) -> Benchmark:
     config = BenchmarkConfig.from_dict(BenchmarkUtils.config_dict_from_id(benchmark_id))
     if config.type == "abstract":
-        return Benchmark(config, None)
-    sys_infos = BenchmarkUtils.load_sys_infos(config)
-    orig_df = BenchmarkUtils.generate_dataframe_from_sys_infos(config, sys_infos)
-    view_dfs = BenchmarkUtils.generate_view_dataframes(config, orig_df)
-    views = [BenchmarkUtils.dataframe_to_table(k, v) for k, v in view_dfs]
-    return Benchmark(config, views)
+        return Benchmark(config, None, None)
+    file_path = benchmark_id + "_benchmark.json"
+    plot_path = benchmark_id + "_plot.csv"
+    benchmark_file = open_cached_file(file_path, datetime.timedelta(days=1))
+    if not benchmark_file:
+        sys_infos = BenchmarkUtils.load_sys_infos(config)
+        orig_df = BenchmarkUtils.generate_dataframe_from_sys_infos(config, sys_infos)
+        view_dfs = BenchmarkUtils.generate_view_dataframes(config, orig_df)
+        json_dict = {k: v.to_dict() for k, v in view_dfs}
+        benchmark_file = os.path.join(get_cache_dir(), sanitize_path(file_path))
+        with open(benchmark_file, "w") as outfile:
+            json.dump(json_dict, outfile)
+        if config.parent:
+            list_data = [
+                json_dict["Demographic-weighted Global Average"]["score"][0],
+                json_dict["Linguistic-weighted Global Average"]["score"][0],
+                datetime.datetime.now(),
+            ]
+            plot_path = os.path.join(get_cache_dir(), sanitize_path(plot_path))
```

Comment:
This is a reasonable idea, but unfortunately it won't work because every time we update the software the cache directory is deleted, so it will reset the plot. I think we'll probably need to follow my previous suggestion of sorting the submissions by date and incrementally calculating the results from the results that only consider earlier systems to those that consider earlier and later systems. I think it's OK to not do this for this particular pull request though, so we could instead create an issue noting this and merge this pull request once it's ready.

Reply:
I have rewritten the code to incrementally calculate results from from the results that only consider earlier systems to those that consider earlier and later systems! (It's definitely more of a rough version, and I can definitely try to make it more efficient :) )